FHEM/14_SD_WS.pm

```diff
@@ -125,6 +127,49 @@ sub SD_WS_Undef {
   my ($hash, $name) = @_;
   carp "SD_WS_Undef, too few arguments ($hash, $name)" if @_ < 2;
   delete($modules{SD_WS}{defptr}{$hash->{CODE}}) if(defined($hash->{CODE}) && defined($modules{SD_WS}{defptr}{$hash->{CODE}}));
+	RemoveInternalTimer($hash);
+  return;
+}
+
+#############################
+sub SD_WS_Set {
+  my ($hash, $name, $cmd, @args) = @_;
+  my $ret = undef;
+	my $LastInputDev = $hash->{LASTInputDev};
```

Comment:
Ich würde spontan behaupten, dass das empfangende IO Device immer in {IODev} zu finden ist.
Da ich kein SD_WS Gerät empfange kann ich es nicht verifizieren, aber Du hast Du vermutlich eines und könntest das bestätigen.

Reply:
IODev gibt es nicht:

```
Internals:
   CODE       SD_WS_33_T_18_2
   DEF        SD_WS_33_T_18_2
   FUUID      61fe7a50-f33f-b27f-ae42-ef51fe7ad7330247
   LASTInputDev sduinoIP
   MSGCNT     23
   NAME       SD_WS_33_T_5
   NR         447
   STATE      Temperatur: 39.0 °C | Trend: rising
   TYPE       SD_WS
   bitMSG     00000100100001001010000111000000000001001000
   lastMSG    0484A1C0048
   lastReceive 1644145627
   sduinoIP_DMSG W33#0484A1C0048
   sduinoIP_MSGCNT 24
   sduinoIP_Protocol_ID 33
   sduinoIP_RAWMSG MS;P0=-2019;P1=494;P5=-3987;P6=-8035;D=1616101010101015101015101010101510101510151010101015151510101010101010101010101510101510;CP=1;SP=6;R=8;
   sduinoIP_RSSI -70
   sduinoIP_TIME 2022-02-06 12:07:07
   sduinoUSB0_DMSG W33#04879DC0044
   sduinoUSB0_MSGCNT 22
   sduinoUSB0_Protocol_ID 33
   sduinoUSB0_RAWMSG MS;P1=-3778;P2=517;P3=-1989;P4=-8039;D=2424232323232321232321232323232121212123232121212321212123232323232323232323232123232321;CP=2;SP=4;m2;
   sduinoUSB0_TIME 2022-02-06 12:03:36
   Helper:
     DBLOG:
       RSSI:
         myDbLog:
           TIME       1644145627.36546
           VALUE      -71
       temperature:
         myDbLog:
           TIME       1644145627.36546
           VALUE      39.0
       temperatureTrend:
         myDbLog:
           TIME       1644145627.36546
           VALUE      rising
   READINGS:
     2022-02-06 11:38:51   MSGCNTLAST10MIN 9
     2022-02-06 12:07:07   channel         2
     2022-02-06 12:07:07   state           T: 39.0
     2022-02-06 12:07:07   temperature     39.0
     2022-02-06 12:07:07   temperatureTrend rising
     2022-02-06 12:07:07   type            E0001PA, s014, S522, TCM, TFA 30.3200, TX-EZ6
Attributes:
   event-min-interval .*:175
   max-deviation-temp 5
   model      S522
   room       SD_WS
   stateFormat Temperatur: temperature °C | Trend: temperatureTrend
```

Reply:
> Ich würde spontan behaupten, dass das empfangende IO Device immer in {IODev} zu finden ist.

Vielleicht meint er das Reading IODev und dort ist NICHT immer das letzte Device drin. Es wird vermutlich nur vom ersten EmpfangsIO gesetzt.

Reply:
Hmm, okay. Wir wollen nichts senden, also müssen wir nicht IODev nehmen? Annahme: IODev automatisch gesetzt, wenn etwas empfangen wird, sofern es registriert wurde:

![image](https://user-images.githubusercontent.com/7968127/152678566-7eff78b6-0322-4e4b-a4c4-719a0ebc9c29.png)

Ist mal wieder so ein Fall, für den es nicht eindeutig geregelt ist, ob man IODev auch fürs Empfangen verwenden soll.  Wir belassen es daher am besten bei LASTInputDev

Reply:
Die Doku ist wahrscheinlich nicht mehr aktuell. Da wurde letztens ziemlich viel am Verhalten von IODev geändert. Zwangsweise gesetzt wird es neuerdings offensichtlich nicht mehr.